os/src/Path.scala

```diff
@@ -388,14 +390,23 @@ object Path {
   }
 
   def apply[T: PathConvertible](f: T, base: Path): Path = apply(FilePath(f), base)
-  def apply[T: PathConvertible](f0: T): Path = {
-    val f = implicitly[PathConvertible[T]].apply(f0)
-    if (f.iterator.asScala.count(_.startsWith("..")) > f.getNameCount/ 2) {
-      throw PathError.AbsolutePathOutsideRoot
+  def apply[T: PathConvertible](f0: T, expandHome: Boolean = true): Path = {
+    val f = {
+      val f = implicitly[PathConvertible[T]].apply(f0)
+      if (f.iterator.asScala.count(_.startsWith("..")) > f.getNameCount/ 2) {
+        throw PathError.AbsolutePathOutsideRoot
```

Comment:
This will throw, when the path contains more `..` elements than path segements, which can be true for relative paths.